backend/src/index.ts

```diff
@@ -136,6 +140,16 @@ class Server {
       }
       await blocks.$updateBlocks();
       await memPool.$updateMempool();
+
+      const blockchainInfo = await bitcoinClient.getBlockchainInfo();
+      if (this.blockIndexingStarted === false
```

Comment:
Maybe move this block of code into the `blocks.ts` file so we can clean up `index.ts` from blockIndexingStarted and bitcoinClient ?